lib/std/heap/general_purpose_allocator.zig

```diff
@@ -155,6 +155,8 @@ pub const Config = struct {
     verbose_log: bool = false,
 };
 
+pub const DeinitResult = enum { success, leaking };
```

Comment:
`deinit()` is the general pattern for every dynamic resource in Zig, so it bears little additional knowledge.
Likewise, `Result` does neither tell a lot on what is actually done.

Something like
```zig
pub const Check = enum { ok, leak };
```
sounds less annoying to type.

External references are then unambiguous as `alloc.check.ok` or `std.heap.check.ok` and this sounds also okayish to write out, if one wants to be explicit.